### PR TITLE
AdverseItem and AdverseAction Resources

### PR DESCRIPTION
**Pull request by rtymchyk**
Addresses issue https://github.com/checkr/checkr-ruby/issues/13

Follows the documentation on:
https://docs.checkr.com/#tag/Adverse-Actions
https://docs.checkr.com/#tag/Adverse-Items

I was not able to implement https://docs.checkr.com/#operation/listOfAdverseActions.
A `GET /v1/adverse_actions` return `Checkr::InvalidRequestError: Resource not found`

Attributing that to outdated API documentation.

### CODE REVIEW

**Comment by rtymchyk** [comment]
@capripot

File: lib/checkr/adverse_item.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+module Checkr
+  class AdverseItem < APIResource
+    attribute :text
+
+    api_class_method :all, :get, '/v1/reports/:report_id/adverse_items', :constructor => APIList.constructor(:AdverseItem)
```
Comment (nothingisfunny) [inline lib/checkr/adverse_item.rb]: api_class_method :all, :get, '/v1/reports/:report_id/adverse_items', :constructor => :AdverseItemList

File: lib/checkr/report.rb [hunk 2]
```diff
@@ -11,6 +11,8 @@ class Report < APIResource
     attribute :candidate, :Candidate
     attribute_writer_alias :candidate_id, :candidate
 
+    attribute :adverse_items, APIList.constructor(:AdverseItem)
```
Comment (nothingisfunny) [inline lib/checkr/report.rb]: attribute :adverse_items, :AdverseItemList, :nested => true, :default => {}
attribute_writer_alias :adverse_item_ids, :adverse_items

Comment (nothingisfunny) [reply lib/checkr/report.rb]: APIList works in a case when the parent object has id/ids of the child attribute. Our public API doesn't return adverse_item_ids as part of the report object, so in this case there is a need to create an `AdverseItemList` (same idea as behind Candidate owning Documents and being able to access them through `DocumentList`). So you'll need to create an additional file `adverse_item_list.rb` with ```module Checkr
  class AdverseItemList < APIList

    attribute :next_href
    attribute :previous_href
    attribute :count
    attr_accessor :parent

    api_instance_method :all, :get

    def self.construct(json, parent=nil)
      lambda = constructor(:AdverseItem)
      instance = lambda.call(json)
      instance.parent = parent if parent
      instance.clear_changed_attributes
      instance
    end

    def path
      parent.path + "/adverse_items"
    end

  end
end```

**Comment by nothingisfunny** [comment]
Hey @rtymchyk! 
Thanks so much for your contribution. I tested your pr locally and found out that the association between Report and AdverseItem doesn't currently work. (I got `adverse_items: null` for a report that had an adverse item, correct me if you had different results when testing it)
I left some comments with changes that made it work for me. And was able to fetch adverse_items belonging to report by calling `Checkr::AdverseItem.all(report_id: report.id)`
Let us know if you could incorporate those in your pr. 
Thanks a lot, 
Lucya

File: lib/checkr/report.rb (hunk 2 above)
Comment (rtymchyk) [reply lib/checkr/report.rb]: Thanks for clarifying, I wasn't sure about the difference between the two.

**Comment by rtymchyk** [comment]
@nothingisfunny Thanks. I was indeed fetching adverse items via `Checkr::AdverseItem.all(report_id: x)`, not through the report resource, so I did not notice it was broken.

Will address the comments hopefully this week.

**Comment by nothingisfunny** [comment]
@rtymchyk Cool, sounds good.

**Comment by rtymchyk** [comment]
@nothingisfunny I've updated the PR with your suggestion.

I can now retrieve Adverse Items in 2 ways:
```ruby
Checkr::Report.retrieve('x').adverse_items.all
```
and
```ruby
Checkr::AdverseItem.all(report_id: 'x')
```

**Comment by rtymchyk** [comment]
There's a failure on Travis which looks related to the environment:
```bash
Could not download 'https://github.com/rvm/rvm/archive/.tar.gz'.
  curl returned status '22'.
Downloading https://bitbucket.org/mpapis/rvm/get/.tar.gz
curl: (22) The requested URL returned error: 404 Not Found
Could not download 'https://bitbucket.org/mpapis/rvm/get/.tar.gz'.
  curl returned status '22'.
Could not update RVM, please report to https://github.com/rvm/rvm/issues
The command "rvm get stable" failed and exited with 22 during .
```

**Comment by rtymchyk** [comment]
Thanks for the quick release.

**Pull request merged**